src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+                                              "https://woogles.io/api/game_service.GameMetadataService/GetGCG "
+                                              "-d '{\"game_id\":\"%s\"}'", game_id_str);
+        
+        char *response = get_process_output(curl_cmd);
+        free(curl_cmd);
+        free(game_id_str);
+        
+        if (response) {
+            // Parse JSON response to extract GCG content
+            const char *gcg_start = strstr(response, "\"gcg\":\"");
+            if (gcg_start) {
+                gcg_start += 7; // Skip "\"gcg\":\""
+                const char *gcg_end = gcg_start;
+                while (*gcg_end && *gcg_end != '"') {
+                    gcg_end++;
+                }
+                int gcg_len = gcg_end - gcg_start;
+                char *raw_gcg = malloc_or_die(gcg_len + 1);
+                strncpy(raw_gcg, gcg_start, gcg_len);
+                raw_gcg[gcg_len] = '\0';
+                
+                // Unescape JSON string using utility function
+                gcg_content = json_unescape_string(raw_gcg, error_stack);
+                free(raw_gcg);
+                
+                if (!gcg_content) {
+                    free(response);
+                    return NULL;
+                }
+            }
+            
+            free(response);
+        }
+    }
+    
+    if (!gcg_content || is_string_empty_or_whitespace(gcg_content) ||
+        strstr(gcg_content, "#player1") == NULL) {
+        free(gcg_content);
+        // Only push error if this was actually a woogles URL
+        if (strstr(identifier, "woogles.io") != NULL) {
+            error_stack_push(error_stack, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND,
+                            get_formatted_string("Failed to download valid GCG from woogles.io: %s",
+                                                  identifier));
+        }
+        return NULL;
+    }
+    
+    return gcg_content;
+}
+
+char *get_local_gcg_string(const char *identifier, ErrorStack *error_stack) {
+    
+    // Try to open as local file
+    FILE *file = fopen(identifier, "r");
+    if (!file) {
+        return NULL;  // Not a valid local file - not our job
+    }
+    fclose(file);
+    
+    // Read file content directly
+    char *gcg_content = get_string_from_file(identifier, error_stack);
+
+    if (!gcg_content || is_string_empty_or_whitespace(gcg_content) ||
+        strstr(gcg_content, "#player1") == NULL) {
+        free(gcg_content);
+        error_stack_push(error_stack, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND,
+                         get_formatted_string("Local file does not contain valid GCG content: %s", identifier));
+        return NULL;
+    }
+
+    return gcg_content;
+}
+
+char *get_gcg_string(const DownloadGCGOptions *options, ErrorStack *error_stack) {
+  
+    if (!options || !options->source_identifier) {
+        error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
```

Comment:
You can make this a log_fatal since it should eventually be impossible for the user to trigger this due to previous checks.